File: Data Structures & Algorithms/duplicate-integer/submission-4.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        a = set(nums)
        if len(a) != len(nums):
            return True
        return False
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
        # numset = set(nums)
        # if len(numset) < len(nums):
        #     return True
        # else: return False

        
        

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    
    //a starting point is only when nums[i] - 1 does not exist
    longestConsecutive(nums) {
        let result = 0
        let numSet = new Set(nums)

        for (let num of numSet){
            let count = 0
            if (numSet.has(num - 1)){
                continue
            }
            else{
                while (numSet.has(num)){
                    count += 1
                    num += 1
                }
                result = Math.max(result, count)
            }
        }
        return result 

    }
}
